src/main/java/mate/academy/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,79 @@
+package mate.academy.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.security.SecurityRequirement;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import mate.academy.dto.cartitem.CartItemDto;
+import mate.academy.dto.cartitem.CartItemRequestDto;
+import mate.academy.dto.cartitem.CartItemUpdateRequestDto;
+import mate.academy.dto.shoppingcart.ShoppingCartDto;
+import mate.academy.service.shoppingcart.ShoppingCartService;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Shopping cart management", description = "Endpoints for managing shopping carts")
+@RestController
+@RequestMapping(value = "/cart")
+@SecurityRequirement(name = "bearerAuth")
+@RequiredArgsConstructor
+public class ShoppingCartController {
+    private final ShoppingCartService shoppingCartService;
+
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @GetMapping
+    @Operation(
+            summary = "Get information about shopping cart",
+            description = "Get all available information about shopping cart")
+    public ShoppingCartDto getAllInfoAboutShoppingCart(Authentication authentication) {
+        return shoppingCartService.getAllInfo(authentication);
```

Comment:
it's better to provide user or userId instead of authentication

Fix other endpoints